From 5b02919a56ed813831ed6656594c0e7228b85f2a Mon Sep 17 00:00:00 2001
From: klzgrad <kizdiv@gmail.com>
Date: Wed, 24 Jan 2018 23:50:06 -0500
Subject: [PATCH] net/spdy: Reduce warnings about RST on invalid streams

Per RFC 7540#6.4:

  However, after sending the RST_STREAM, the sending endpoint MUST be
  prepared to receive and process additional frames sent on the stream
  that might have been sent by the peer prior to the arrival of the
  RST_STREAM.
---
 src/net/spdy/spdy_session.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/net/spdy/spdy_session.cc b/src/net/spdy/spdy_session.cc
index db8fcac08..02b44a6ca 100644
--- a/src/net/spdy/spdy_session.cc
+++ b/src/net/spdy/spdy_session.cc
@@ -3140,7 +3140,7 @@ void SpdySession::OnWindowUpdate(spdy::SpdyStreamId stream_id,
 
     if (it == active_streams_.end()) {
       // NOTE:  it may just be that the stream was cancelled.
-      LOG(WARNING) << "Received WINDOW_UPDATE for invalid stream " << stream_id;
+      VLOG(1) << "Received WINDOW_UPDATE for invalid stream " << stream_id;
       return;
     }
 
@@ -3185,7 +3185,7 @@ void SpdySession::OnHeaders(spdy::SpdyStreamId stream_id,
   auto it = active_streams_.find(stream_id);
   if (it == active_streams_.end()) {
     // NOTE:  it may just be that the stream was cancelled.
-    LOG(WARNING) << "Received HEADERS for invalid stream " << stream_id;
+    VLOG(1) << "Received HEADERS for invalid stream " << stream_id;
     return;
   }
 
-- 
2.53.0

